File: 1114-binary-search-tree-to-greater-sum-tree/binary-search-tree-to-greater-sum-tree.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    int output = 0;
    public TreeNode BstToGst(TreeNode root) {
        if(root != null){
            BstToGst(root.right);
            output += root.val;
            root.val = output;
            BstToGst(root.left);
        }
        return root;
    }
}

File: 1114-binary-search-tree-to-greater-sum-tree/binary-search-tree-to-greater-sum-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def __init__(self):
        self.output = 0
    def bstToGst(self, root: TreeNode) -> TreeNode:
        if root:
            self.bstToGst(root.right)
            self.output += root.val
            root.val = self.output
            self.bstToGst(root.left)
        return root
        